srcs/execution/builtins/echo.c

```diff
@@ -0,0 +1,139 @@
+#include "builtins.h"
+#include "opt.h"
+#include "libft.h"
+
+/*
+** for every arguments:
+** '-e': backslash escapes interpretation
+** handle '-s' and '-n' options
+*/
+
+int		builtin_echo(int argc, char **argv)
+{
+	int		i;
+	t_opt	*o;
+
+	o = (t_opt*)malloc(sizeof(t_opt));
+	i = retrieve_options(argc, argv, o);
+	while (++i < argc)
+	{
+		if (o->e == true)
+			escape_char(argv[i]);
+		else
+			ft_putstr(argv[i]);
+		if (o->s == false)
+			ft_putchar(' ');
+	}
+	if (o->n == false)
+		ft_putchar('\n');
+	free(o);
+	return (STATUS_SUCCESS);
+}
+
+/*
+** Parcour every char
+** if start with '\':
+** display one char '\x' for two chars "\x"
+** increment index to not display escaped chars
+*/
+
+void	escape_char(char *str)
+{
+	int	i;
+
+	i = -1;
+	while (str[++i] != '\0')
+	{
+		if (str[i] == '\\' && str[i + 1])
+		{
+			escape(*(&str[++i]));
+			if (str[i] == '0')
+				i += octal(&str[i]);
+		}
+		else
+			ft_putchar(str[i]);
+	}
+}
+
+/*
+** '\c' is not below as it display nothing
```

Comment:
Nice, thanks for the test.
I did not understand the `\c` escape stops displaying.
It's now implemented!